embark-ui/src/sagas/searchSaga.js

```diff
@@ -1,16 +1,28 @@
 import {put, select} from "redux-saga/effects";
-import {getAccounts, getBlocks, getTransactions, getContracts} from "../reducers/selectors";
-import {fetchAccounts, fetchBlocks, fetchTransactions, fetchContracts} from "./index";
+import {getAccounts, getBlocks, getTransactions, getContracts, getEnsRecordForName} from "../reducers/selectors";
+import {fetchAccounts, fetchBlocks, fetchTransactions, fetchContracts, fetchEnsRecord} from "./index";
 import {ELEMENTS_LIMIT} from '../constants';
 
 export function *searchExplorer(entity, payload) {
   let result;
+  let searchValue = payload.searchValue;
+  let isENSName = false;
+  if (searchValue.endsWith('.eth')) {
```

Comment:
What if I configure my root domain to be `.luri`?

I guess the only things we could do is to check if it is something that look like ens name:
string separated with dot.

Reply:
You're a bit dumb if you configure as `.luri` lol.
But I agree that just supporting `.eth` is not the best, especially since support for `.xyz` is coming.
There is a lot of places in the code where we make the assumption of `.eth`, so it should be a bigger effort that just adjusting here.
IMO, the ENS module should have a list of accepted extensions and throw if you enter something out of it. Other modules that check for `.eth` would be changed too to use the same list of accepted extensions, What do you think?

Reply:
`There is a lot of places in the code where we make the assumption of .eth`
=> REALLY!!!

I agree with what you suggest a whitelist of valid extension seems totally reasonable

Reply:
Well, maybe not "a lot", but I know there's a least one other place haha.
I'll do those changes in another PR. Will open a Pivotal ticket.